### ISSUE TITLE
Record glitch timestamps

### ISSUE THREAD

**Issue by alecgunny** [ML4GW/BBHNet#302]
When we break up training and validation glitches at the moment, we don't have any information about the timestamp of each glitch and naively break them up according to `valid_frac`. But this leaves the possibility that we have glitches from the training set in the validation set, which can throw off our estimates.

More broadly, what I've realized in the course of doing #295 is that it really doesn't make sense to measure performance against a specific set of glitch predictions: those predictions are already contained in the validation set (if we implement the correct way of doing things above), and so if those glitches are truly the most problematic, they should bubble up to the top of the list when we do e.g. recall or AUROC at various thresholds that just compare signal predictions to the most problematic events.

**Comment by alecgunny** [comment]
In the context of my second point above, we'd still want to record timestamps to make sure that no glitches from the validation period end up in the training set.
